pkg/fqdn/service/service.go

```diff
@@ -344,9 +348,18 @@ func (s *FQDNDataServer) sendAndRecvAckForDNSPolicies(stream pb.FQDNData_StreamP
 					continue
 				}
 
+				endpointID := uint32(ep.GetID())
+
+				// Skip if we've already added this endpoint for this DNS policy
```

Comment:
optional nit: note that a `if sets.Set.Add` would make this even shorter

Reply:
Not seeing an `Add` operation for the Set. I do see an `Insert` though ? is that what you are referring to. That would be similar to this I think. We will need to have `Has(endpointID)` for the set.
https://pkg.go.dev/k8s.io/apimachinery/pkg/util/sets